nltk/metrics/confusionmatrix.py

```diff
@@ -61,7 +61,7 @@ def key(v):
         indices = {val: i for (i, val) in enumerate(values)}
 
         # Make a confusion matrix table.
-        confusion = [[0] * len(values)] * len(values)
+        confusion = [[0 for val in values] for val in values]
```

Comment:
Or perhaps:
```confusion = [[0 for _ in values] for _ in values]```

Reply:
cf https://github.com/nltk/nltk/pull/2939/commits/18bf2bd4f87034632903c687e23cf982f177a33f

Reply:
Well spotted - that is better.

Reply:
NB the literal list is replicated as expected but the resulting list is then referenced multiple times.

Reply:
Indeed.
For the record, the original is this:
https://github.com/nltk/nltk/blob/0fac0c0f8e4618c2bdd3d2137d5fb8a80f581246/nltk/metrics/confusionmatrix.py#L64

However, GitHub shows the original unintuitively. The timeline is as follows:
* The original is as above.
* I carelessly turn it into `[[0] * len(values)] * len(values)` in 024b490b6c2cc6a70c06cb64d1b6760daa805151
* I realise that this change produces incorrect outputs, so I revert this commit in 21e20d041641918d87262a03e33f47213792289f. The comment that you're currently reading is made on this commit, hence the "wrong" original.
* I submit the PR.

And yes, `[[0] * len(values)] * len(values)` is buggy:
```python
>>> from nltk.metrics import ConfusionMatrix
>>> ref  = 'DET NN VB DET JJ NN NN IN DET NN'.split()
>>> test = 'DET VB VB DET NN NN NN IN DET NN'.split()
>>> cm = ConfusionMatrix(ref, test)
    | D         |
    | E I J N V |
    | T N J N B |
----+-----------+
DET |<3>1 . 4 2 |
 IN | 3<1>. 4 2 |
 JJ | 3 1<.>4 2 |
 NN | 3 1 .<4>2 |
 VB | 3 1 . 4<2>|
----+-----------+
(row = reference; col = test)
```

It should be all good now.